File: JAVA/PROGRAMACION_ANDROID_1/LoginConRegistro/app/src/main/java/com/david/loginconregistro/MainActivity.java
package com.david.loginconregistro;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.textfield.TextInputLayout;

public class MainActivity extends AppCompatActivity {

    // SE DECLARAN LOS LEMENTOS UTILIZADOS
    Button BtnOk;
    TextView TvRegister;
    TextInputLayout tilUser, tilPass;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // VER SI SE HA INICIADO ANTES UNA SESIÓN
        comprobar();

        // ENLAZAR LOS CONTROLADORES DE LA VISTA
        BtnOk = (Button) findViewById(R.id.BtnOk);
        tilUser = (TextInputLayout) findViewById(R.id.tilUser);
        tilPass = (TextInputLayout) findViewById(R.id.tilPass);
        TvRegister = (TextView) findViewById(R.id.TvRegister);

        // PROGRAMAR EL OYENTE PARA ACCEDER
        BtnOk.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Acceder();
            }
        });

        // SE PROGRAMA EL OYENTE PARA LA TEXT VIEW DE REGISTRAR
        TvRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                registrar();
            }
        });

        // SE CREA EL SHARED PREFERENCES
        CrearSharedPreferences();
    }

    // SE PROGRAMA LA FUNCIÓN
    private void Acceder(){
        // SE INTANCIA EL SHARED
        SharedPreferences sp1 = getSharedPreferences("MisPreferencias", Context.MODE_PRIVATE);

        // DECLARAMOS VARIABLES PARA COMPARAR VALORES
        String u = tilUser.getEditText().getText().toString();
        String p = tilPass.getEditText().getText().toString();

        // VALIDAMOS QUE NO ESTEN VACÍOS LOS CAMPOS
        if (u.isEmpty() || p.isEmpty()) {
            Toast.makeText(getApplicationContext(), "NO DEJES NINGÚN CAMPO VACÍO", Toast.LENGTH_LONG).show();
        } else {
            // COMPARAMOS LO DE LA CAJAS DE TEXTO CON EL SHARED PREFERENCES
            // OBTENER LOS DATOS QUE YA SE HAN ALMACENADO
            String usuario = sp1.getString("usuario", "NO SE ENCONTRÓ EL USUARIO");
            String password = sp1.getString("password", "NO SE ENCONTRÓ LA CONTRASEÑA");

            if (u.equals(usuario) && p.equals(password)) {
                // AGREGAMOS UN TOAST PARA QUE NOS DIGA BIENVENIDO XD
                Toast.makeText(getApplicationContext(), "BIENVENIDO "+usuario, Toast.LENGTH_LONG).show();

                // SE INSTANCIA EL ACTIVITY               contexto, nombre de la clase compilada
                Intent intent1 = new Intent(getApplicationContext(), InicioActivity.class);

                // SE MANDA A EJECUTAR EL ACTIVITY
                startActivity(intent1);
            } else {
                // MUESTRA UN TOAST EN CASO DE QUE LOS DATOS NO SEAN IGUAL A LOS DEL SHARED PREFERENCES
                Toast.makeText(getApplicationContext(), "DATOS INCORRECTOS", Toast.LENGTH_SHORT).show();
            }
        }
    }

    // SE CREA LA FUNCIÓN DEL SHARED PREFERENCES
    private void CrearSharedPreferences(){
        // SE INSTANCIA EL SHARED                   Nombre del archivo,   forma en como se abrirá
        SharedPreferences prefer1 = getSharedPreferences("MisPreferencias", Context.MODE_PRIVATE);

        // SE ACTIVA LA EDICIÓN DEL ARCHIVO
        SharedPreferences.Editor editor = prefer1.edit();

        // SE MANDAN LOS VALORES QUE QUEREMOS ALMACENAR
        // Nombre de la variable, valor de la variable
        editor.putString("usuario", "David");
        editor.putString("password", "102030");

        // PARA CONTROLAR LOS CIERRES DE SESIÓN
        editor.putBoolean("ingreso", false);

        // SE MANDA UNA INSTRUCCIÓN COMMIT PARA QUE SE GUARDEN LOS CAMBIOS
        editor.commit();
    }

    // METODO QUE VERIFICA SI SE HA CERRADO O NO UNA SESION
    private void comprobar(){
        // ACTIVAMOS EL EDITOR DEL SHARED
        SharedPreferences sp1 = getSharedPreferences("MisPreferencias", Context.MODE_PRIVATE);

        // TRAEMOS EL VALOR DE LA SESION
        boolean val = sp1.getBoolean("ingreso", false);
        String usuario = sp1.getString("usuario", "NO SE ENCONTRÓ EL USUARIO");

        if(val==true) {
            // AGREGAMOS UN TOAST PARA QUE NOS DIGA BIENVENIDO XD
            Toast.makeText(getApplicationContext(), "BIENVENIDO "+usuario, Toast.LENGTH_LONG).show();

            // SE INSTANCIA EL ACTIVITY               contexto, nombre de la clase compilada
            Intent intent1 = new Intent(getApplicationContext(), InicioActivity.class);

            // SE MANDA A EJECUTAR EL ACTIVITY
            startActivity(intent1);
        }
    }

    public void registrar(){
        // SE INSTANCIA EL ACTIVITY
        Intent intent2 = new Intent(getApplicationContext(), RegisterActivity.class);

        // SE MANDA A EJECUTAR EL ACTIVITY
        startActivity(intent2);
    }
}

File: KOTLIN/E1-Calculadora_Basica_Kotlin_DMM/app/src/main/java/com/example/androidsample/MainActivity.kt
package com.example.androidsample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var operations: Operations
    private var num1 = 0
    private var num2 = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        operations = Operations()
    }

    fun convertCurrency(view : View){
        if(number1.text.isNotEmpty()){
            val dollarValue = number1.text.toString().toFloat()
            val euroVaue = dollarValue * 0.85f
            textView.text = euroVaue.toString()
        } else {
            textView.text = getString(R.string.no_value_string)
        }
    }

    fun addOpe(view: View){
        if(validityInputs()){
            num1 = number1.text.trim().toString().toInt()
            num2 = number2.text.trim().toString().toInt()
            textView.text = operations.add(num1, num2).toString()
        } else {
            textView.text = getString(R.string.no_value_string)
        }
    }

    fun substractOpe(view: View){
        if(validityInputs()){
            num1 = number1.text.trim().toString().toInt()
            num2 = number2.text.trim().toString().toInt()
            textView.text = operations.subtract(num1, num2).toString()
        } else {
            textView.text = getString(R.string.no_value_string)
        }
    }

    fun multiOpe(view: View){
        if(validityInputs()){
            num1 = number1.text.trim().toString().toInt()
            num2 = number2.text.trim().toString().toInt()
            textView.text = operations.multiply(num1, num2).toString()
        } else {
            textView.text = getString(R.string.no_value_string)
        }
    }

    fun divideOpe(view: View){
        if(validityInputs()){
            num1 = number1.text.trim().toString().toInt()
            num2 = number2.text.trim().toString().toInt()
            textView.text = operations.divide(num1, num2).toString()
        } else {
            textView.text = getString(R.string.no_value_string)
        }
    }

    fun validityInputs(): Boolean{
        if(number1.text.toString().isNotEmpty() && number2.text.toString().isNotEmpty()){
            return true
        }
        return false
    }
}